core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -129,7 +130,7 @@ public TableMetadata refresh() {
 
   @Override
   public void commit(TableMetadata base, TableMetadata metadata) {
-    Pair<Integer, TableMetadata> current = versionAndMetadata();
+    Pair<Long, TableMetadata> current = versionAndMetadata();
```

Comment:
The change from int to long seems unrelated?

Reply:
Yes, I just thought that versionId defined as integer type is easy to overflow, so I changed it to long type.